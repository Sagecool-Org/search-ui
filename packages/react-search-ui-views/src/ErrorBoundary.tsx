import React from "react";

import { appendClassName } from "./view-helpers";
import type { SearchContextState } from "@elastic/search-ui";
import { BaseContainerProps } from "./types";

export type ErrorBoundaryContainerContext = Pick<SearchContextState, "error">;

export type ErrorBoundaryViewProps = BaseContainerProps &
  ErrorBoundaryContainerContext;

function ErrorBoundary({
  children,
  className,
  error,
  ...rest
}: ErrorBoundaryViewProps &
  React.HTMLAttributes<HTMLDivElement>): React.JSX.Element {
  if (error) {
    return (
      <div className={appendClassName("sui-search-error", className)} {...rest}>
        {error}
      </div>
    );
  }

  return children as React.JSX.Element;
}

export default ErrorBoundary;
