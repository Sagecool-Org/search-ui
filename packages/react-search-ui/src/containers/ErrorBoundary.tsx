import React from "react";
import {
  ErrorBoundary,
  ErrorBoundaryViewProps
} from "@elastic/react-search-ui-views";
import { BaseContainerProps } from "../types";
import { useSearch } from "../hooks";

type ErrorBoundaryContainerProps = BaseContainerProps & {
  view?: React.ComponentType<ErrorBoundaryViewProps>;
};
export const ErrorBoundaryContainer = ({
  children,
  className,
  view,
  ...rest
}: ErrorBoundaryContainerProps) => {
  const { error } = useSearch();
  const View = view || ErrorBoundary;

  const viewProps = {
    className,
    children,
    error,
    ...rest
  };

  return <View {...viewProps} />;
};

export default ErrorBoundaryContainer;
