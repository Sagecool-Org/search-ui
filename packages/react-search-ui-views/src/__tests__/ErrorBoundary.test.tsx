import React from "react";
import { render } from "@testing-library/react";
import ErrorBoundary from "../ErrorBoundary";

const params = {
  children: <div>Child</div>,
  error: "I am an error"
};

it("renders an error when there is an error", () => {
  const { container } = render(<ErrorBoundary {...params} />);
  expect(container).toMatchSnapshot();
});

it("renders children when there is no error", () => {
  const { container } = render(
    <ErrorBoundary
      {...{
        ...params,
        error: ""
      }}
    />
  );
  expect(container).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const { container } = render(
    <ErrorBoundary className={customClassName} {...params} />
  );
  expect(container.firstChild).toHaveClass("sui-search-error", "test-class");
});
