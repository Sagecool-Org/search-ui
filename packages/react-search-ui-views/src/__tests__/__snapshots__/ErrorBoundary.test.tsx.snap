// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an error when there is an error 1`] = `
<div>
  <div
    class="sui-search-error"
  >
    I am an error
  </div>
</div>
`;

exports[`renders children when there is no error 1`] = `
<div>
  <div>
    Child
  </div>
</div>
`;
